### PR TITLE
Update gd.py to reduce recommit_particles calls

### PR DESCRIPTION
**Pull request by rieder**
Allow set_* methods from UPDATE state without going through RUN state, reducing the amount of recommit_particles calls

### CODE REVIEW

**Comment by rieder** [comment]
Test below should have only one call to recommit_particles, before this patch there are two.

```python
import logging
from amuse.lab import *

logging.basicConfig(level=logging.DEBUG)
logging.getLogger("code").setLevel(logging.DEBUG)

particles = new_plummer_sphere(100)

grav = Hermite(redirection="none")
grav.particles.add_particles(particles)
grav.evolve_model(0.01 | nbody_system.time)

channel = particles.new_channel_to(grav.particles)
grav_attributes = ["x", "vx"]

particles.x += 1 | nbody.length
particles.vx += 1 | nbody.speed

channel.copy_attributes(grav_attributes)
grav.evolve_model(0.02 | nbody_system.time)

grav.stop()
```

**Comment by ipelupessy** [comment]
hmm, the script you provide is not working here (doesn't want to set "x" and "vx" only)

**Comment by ipelupessy** [comment]
I have enabled/added logging; you can add more logging in the state if you wish..

**Comment by rieder** [comment]
> hmm, the script you provide is not working here (doesn't want to set "x" and "vx" only)

Odd. I initially tested with xyz and vxyz, changed it now to do that again (I thought just x and y would simplify it, weird that it doesn't work!).

**Comment by ipelupessy** [comment]
its a limitation from the fact that set_position sets all three position attributes at once...its a bit weird that "position" also doesn't work...

**Comment by ipelupessy** [comment]
I think its ok to merge btw, there may be other problems (it could use a bit more tests), but with the better logging hopefully it will be easier to resolve these...

**Comment by rieder** [comment]
position is a read_only version of [x,y,z] I think... so it can't be set.

File: src/amuse/support/state.py [hunk 1]
```diff
@@ -139,11 +138,7 @@ def __init__(self, interface, **options):
     @option(type='boolean', sections=['state',])
     def is_enabled(self):
         return True
-        
-    @option(type='boolean', sections=['state',])
```
Comment (rieder) [inline src/amuse/support/state.py:143]: why remove this? maybe just default to True instead?

Comment (ipelupessy) [reply src/amuse/support/state.py:143]: its just duplicating logging functionality..we also don't do it when logging the channel stuff...

**Pull request merged**